src/endpoints/authentication/jwt_auth.cpp

```diff
@@ -129,23 +128,55 @@ namespace ccf
     }
 
     auto& token = token_opt.value();
-    auto keys = tx.ro<JwtPublicSigningKeys>(
+    auto keys = tx.ro<JwtPublicSigningKeysMetadata>(
       ccf::Tables::JWT_PUBLIC_SIGNING_KEYS_METADATA);
     const auto key_id = token.header_typed.kid;
     auto token_keys = keys->get(key_id);
 
+    // For metadata KID->(cert,issuer,constraint).
+    //
+    // Note, that Legacy keys are stored as certs, new approach is raw keys, so
+    // conversion from cert to raw key is needed.
+    if (!token_keys)
+    {
+      auto fallback_certs = tx.ro<JwtPublicSigningKeysMetadataLegacy>(
+        ccf::Tables::Legacy::JWT_PUBLIC_SIGNING_KEYS_METADATA);
+      auto fallback_data = fallback_certs->get(key_id);
+      if (fallback_data)
+      {
+        auto new_keys = std::vector<OpenIDJWKMetadata>();
+        for (const auto& metadata : *fallback_data)
+        {
+          auto verifier = ccf::crypto::make_unique_verifier(metadata.cert);
+          new_keys.push_back(OpenIDJWKMetadata{
+            .public_key = verifier->public_key_der(),
+            .issuer = metadata.issuer,
+            .constraint = metadata.constraint});
+        }
+        if (!new_keys.empty())
+        {
+          token_keys = new_keys;
+        }
+      }
+    }
+
+    // For metadata as two separate tables, KID->JwtIssuer and KID->Cert.
+    //
+    // Note, that Legacy keys are stored as certs, new approach is raw keys, so
+    // conversion from certs to keys is needed.
     if (!token_keys)
     {
       auto fallback_keys = tx.ro<Tables::Legacy::JwtPublicSigningKeys>(
         ccf::Tables::Legacy::JWT_PUBLIC_SIGNING_KEYS);
       auto fallback_issuers = tx.ro<Tables::Legacy::JwtPublicSigningKeyIssuer>(
         ccf::Tables::Legacy::JWT_PUBLIC_SIGNING_KEY_ISSUER);
 
-      auto fallback_key = fallback_keys->get(key_id);
-      if (fallback_key)
+      auto fallback_cert = fallback_keys->get(key_id);
+      if (fallback_cert)
       {
+        auto verifier = ccf::crypto::make_unique_verifier(*fallback_cert);
```

Comment:
Not efficient, but not important for a fallback path, will get replaced with RSA key retrieval directly in #6673, can't do it now because adding `rsa_` headers breaks the compilation :/